### PR TITLE
[eclipse/xtext#1679] converted Xtend code to Java

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/xtext#1679] converted Xtend code to Java
Signed-off-by: Christian Dietrich <christian.dietrich@itemis.de>

### CODE REVIEW

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/hierarchy/AbstractHierarchyBuilder.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+			}
+		}
+		return null;
+	}
+
+	protected IEObjectDescription getDescription(EObject object) {
+		if (object == null) {
+			return null;
+		}
+		return IterableExtensions.head(getIndexData().getExportedObjectsByObject(object));
```
Comment (nbhusare) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/hierarchy/AbstractHierarchyBuilder.java:80]: You might like to get rid of `IterableExtensions`
```
Iterable<IEObjectDescription> exportecObjects = getIndexData().getExportedObjectsByObject(object)
return exportecObjects.hasNext() ? exportecObjects.next() : null;
```

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/hierarchy/DefaultHierarchyNodeLocationProvider.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		ITextRegion textRegion = locationInFileProvider.getSignificantTextRegion(owner, feature, indexInList);
+		return toTextRegionWithLineInformation(owner, textRegion);
+	}
+
+	protected ITextRegionWithLineInformation toTextRegionWithLineInformation(EObject obj, ITextRegion textRegion) {
+		if (textRegion == null) {
+			return ITextRegionWithLineInformation.EMPTY_REGION;
+		}
+		if (textRegion instanceof ITextRegionWithLineInformation) {
+			return ((ITextRegionWithLineInformation) textRegion);
```
Comment (nbhusare) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/hierarchy/DefaultHierarchyNodeLocationProvider.java:55]: `return (ITextRegionWithLineInformation) textRegion;`

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/refactoring/ResourceRelocationContext.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+import org.eclipse.emf.ecore.resource.ResourceSet;
+import org.eclipse.xtend.lib.annotations.FinalFieldsConstructor;
+import org.eclipse.xtext.ide.serializer.IChangeSerializer;
+import org.eclipse.xtext.xbase.lib.Exceptions;
+
+/**
+ * @author koehnlein - Initial contribution and API
+ * @since 2.13
+ */
+@FinalFieldsConstructor
```
Comment (nbhusare) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/refactoring/ResourceRelocationContext.java:24]: Please remove the annotation, and also the import.

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/hierarchy/AbstractHierarchyBuilder.java (hunk 1 above)
Comment (cdietrich) [reply org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/editor/hierarchy/AbstractHierarchyBuilder.java:80]: will keep for now.

File: org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/refactoring/ResourceRelocationContext.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+	 * @param change
+	 *            the change to execute
+	 */
+	protected Resource loadAndWatchResource(ResourceRelocationChange change) {
+			if (changeType != null) {
+				switch (changeType) {
+				case MOVE:
+				case RENAME:
+					Resource original = resourceSet.getResource(change.getFromURI(), true);
+					changeSerializer.addModification(original, (Resource it) -> {
```
Comment (nbhusare) [inline org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/refactoring/ResourceRelocationContext.java:67]: `changeSerializer.addModification(original, (Resource it) -> original.setURI(change.getToURI()));

**Pull request merged**